File: test/fixtures/cpp/DISample.cpp
#include <vector>
#include "UserService.h"
#include "UserRepository.h"

class DISample {
private:
    UserService* userService;
    std::vector<UserRepository> repositories;

public:
    DISample(UserService* service, std::vector<UserRepository> repos) 
        : userService(service), repositories(repos) {
    }

    void doWork() {
        userService->process();
    }
};

File: test/fixtures/java/DISample.java
package com.example.di;

import java.util.List;
import com.example.services.UserService;
import com.example.repositories.UserRepository;

public class DISample {
    private final UserService userService;
    private final List<UserRepository> repositories;

    public DISample(UserService userService, List<UserRepository> repositories) {
        this.userService = userService;
        this.repositories = repositories;
    }

    public void doWork() {
        userService.process();
    }
}
